api/numaresourcesoperator/v1/numaresourcesoperator_types.go

```diff
@@ -22,6 +22,7 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/openshift-kni/numaresources-operator/internal/api/features"
```

Comment:
let's avoid this, will make the package non importable: https://docs.google.com/document/d/1e8kOo3r51b2BWtTs_1uADIA5djfXhPT36s6eHVRIvaU/edit

Reply:
Here: https://github.com/openshift-kni/numaresources-operator/pull/1025#discussion_r1784578995 you mentioned you want it non-importable, I'm a bit confused :)

Reply:
I could have been clearer. The magic annotations should not be part of the API packages, nor of its helpers.
Thus, let's move the helpers away from the api packages: rather than import the internal package in the api package, let's move the helper in the internal package.